### ISSUE TITLE
Failed to create directory

### ISSUE THREAD

**Issue by dschinkel** [theohbrothers/ConvertOneNote2MarkDown#180]
ugh so it was chugging along fine and then the below.  Not sure why it's always stopping at my Music OneNote page and then after a few more keeps going just fine.  Not sure what's wrong with these pages.

I don't see anything particularly troubling looking at my music pag so not sure what path is not "legal" and what that means to the script.

When I go look at the output, it actually looks like most of it was indeed exported, so not sure which parts failed here and what.

<img width="767" alt="music" src="https://github.com/theohbrothers/ConvertOneNote2MarkDown/assets/338798/d404ca45-1380-443a-9548-a43154858004">

```
Failed to create directory \
Failed to convert page: Projects\Music
The path is not of a legal form.
at Convert-OneNotePage<Process>, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1107
at New-SectionGroupConversionConfig, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1025
at Convert-OneNote2MarkDown, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1435
at <ScriptBlock>, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1471
at <ScriptBlock>, <No file>: line 1
Failed to create directory
```

```
### Misc Ideas [Page]
pdf file ready: \\?\C:\users\coffeegrindsWindows\Downloads\converted-onenote\PERSONAL\Projects\Misc-Ideas.pdf
Markdown file ready: C:\users\coffeegrindsWindows\Downloads\converted-onenote\PERSONAL\Projects\Misc-Ideas.md
## Building conversion configuration for Music [Page]
### Music [Page]
Convert-OneNotePage : Failed to create directory \
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage

Convert-OneNotePage : Failed to convert page: Projects\Music
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage

Convert-OneNotePage : The path is not of a legal form.
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage

Convert-OneNotePage : at Convert-OneNotePage<Process>, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line
1107
at New-SectionGroupConversionConfig, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1025
at Convert-OneNote2MarkDown, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1435
at <ScriptBlock>, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1471
at <ScriptBlock>, <No file>: line 1
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage

## Building conversion configuration for 221 [Page]
### 221 [Page]
Convert-OneNotePage : Failed to create directory
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage

Convert-OneNotePage : Failed to convert page: Projects\221
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage

Convert-OneNotePage : Cannot bind argument to parameter 'Path' because it is an empty string.
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage

Convert-OneNotePage : at Convert-OneNotePage<Process>, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line
1107
at New-SectionGroupConversionConfig, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1025
at Convert-OneNote2MarkDown, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1435
at <ScriptBlock>, C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1: line 1471
at <ScriptBlock>, <No file>: line 1
At C:\users\coffeegrindsWindows\Downloads\ConvertOneNote2MarkDown-master\ConvertOneNote2MarkDown-v2.ps1:1435 char:257
+ ... onConfigs | Convert-OneNotePage -OneNoteConnection $OneNote -Config $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Convert-OneNotePage
```

**Comment by leojonathanoh** [comment]
are you using powershell ISE? Please use Windows Powershell, and not ISE, because it does not support long paths, see FAQ.

**Comment by dschinkel** [comment]
No I'm not.  But noticed that after I went into File | Options | Advanced in OneNote and set it to lock after 1 hour and not 5 min for password protected pages, then unprotected them with my pwd, then ran this, all looks smooth now.

**Comment by leojonathanoh** [comment]
Good to know. It's quite a mess with Onenote, I understand all the hoops to get through this, haven't used password protection but hope your case helps future readers. 😄

**Comment by dschinkel** [comment]
Yea thanks a ton.  Even though it'll still be a little manual importing this into Notion, it's going to allow me to get the hell out of MS who keeps preventing you from leaving due to lack of export capability.  This is the last of MS, and I'll be free from paying those bastards!

**Comment by dschinkel** [comment]
I may eventually make a video on my channel showing how to use the script https://www.youtube.com/@dimecasts

**Comment by dschinkel** [comment]
In terms of importing into Notion, I found the best way was to import via the generated Word docs in the docx folder that is created for a given notebook.  It retained the formatting better including kept the images in tact in the pages in the word doc.  When I used the .md, Notion didn't translate some things and kept some things wrapped in HTML tags.  Might be a matter of picking the right markup export format, but I just tried the docx and it worked great so didn't spend time trying to figure out which md format Notion liked best.

There is a limit though with Notion, an import file cannot exceed 5mb so that is one obstacle to figure out how to get around.  You'll have to split files into multiple if they exceed 5mb.  No biggie.
